src/Storages/StorageDistributed.cpp

```diff
@@ -946,7 +1057,10 @@ QueryTreeNodePtr buildQueryTreeDistributed(SelectQueryInfo & query_info,
     const StorageSnapshotPtr & distributed_storage_snapshot,
     const StorageID & remote_storage_id,
     const ASTPtr & remote_table_function,
-    const ASTPtr & additional_filter = nullptr)
+    const ASTPtr & additional_filter = nullptr,
+    const NameSet * columns_to_cast = nullptr,
+    const ColumnsDescription * metadata_columns = nullptr,
+    const ColumnsDescription * actual_columns_for_cast = nullptr)
```

Comment:
Looks like it can't be null in both calls.

Reply:
Perhaps create an overload that receives less parameters and call this one with nullptr to avoid excessive default arguments